pkg/service/volume_service_test.go

```diff
@@ -87,53 +140,116 @@ func (suite *VolumeServiceTestSuite) getDefaultVolumeOptions() *nimbleos.Volume
 		LimitMbps:      &limitMbpsField,
 		MultiInitiator: param.NewBool(true),
 		AgentType:      nimbleos.NsAgentTypeNone,
+		FolderId:		folder.ID,
 	}
 	return newVolume
 }
 
-func (suite *VolumeServiceTestSuite) createVolume(volumeName string) *nimbleos.Volume {
-	volume, _ := suite.volumeService.GetVolumeByName(volumeName)
+func (suite *VolumeServiceTestSuite) nonTenantCreateVolume(volumeName string) *nimbleos.Volume {
+	// Create a volume for non tenant
+	volume, _ := suite.nonTenantVolumeService.GetVolumeByName(volumeName)
```

Comment:
If we reuse an existing volume, it might not have been created by a tenant.  Maybe we aren't testing what we mean to test?  Similar thing in some of the other functions.

Reply:
mm I'm not sure what you mean.. There's 2 different createVolume functions: nonTenantCreateVolume() and tenantCreateVolume() so we can be sure which user does what

Reply:
If a volume with the requested name exists before the "create" function is called, a new volume is not created.  So, the test might not verify that a tenant can create a volume.

Reply:
ah okk. Just to double check -- you're referring to volume creation in SetupTest(), right?

Reply:
The volume creation is invoked in SetupTest(), yes, but that's not where I'm focusing.  Presumably the reason there are separate functions for tenant and non-tenant creation of volumes is to verify the behavior in both cases.  And it's not clear that is what is happening.  Maybe the GetVolumeByName() calls aren't needed in those functions, or maybe a pre-existing volume should cause a test failure.

Reply:
got it!